src/feature_scaling.jl

```diff
@@ -1,57 +1,146 @@
 """
-`μ = center!(X[, μ])`
+`μ = center!(X[, μ, obsdim])`
 
-Centers each row of `X` around the corresponding entry in the vector `μ`.
+Centers `X` along obsdim around the corresponding entry in the vector `μ`.
 If `μ` is not specified then it defaults to `mean(X, 2)`.
 """
-function center!{T<:AbstractFloat}(X::AbstractMatrix{T}, μ::AbstractVector = vec(mean(X, 2)))
-    k, n = size(X)
-    @inbounds for j = 1:k
-        for i = 1:n
-            X[j, i] = X[j, i] - μ[j]
+function center!(X, μ; obsdim=LearnBase.default_obsdim(X))
+    center!(X, μ, LearnBase.obs_dim(obsdim))
```

Comment:
can you change 

```julia
LearnBase.obs_dim(obsdim)
```

everywhere to

```julia
convert(LearnBase.ObsDimension, obsdim)
```

(I deprecated the former just a few days ago)